website/src/controllers/top-level/payments/amazon.js

```diff
@@ -0,0 +1,280 @@
+/* import async from 'async';
+import cc from 'coupon-code';
+import mongoose from 'mongoose';
+import moment from 'moment';
+import payments from './index';
+import shared from '../../../../../common';
+import { model as User } from '../../../models/user'; */
+import {
+  // NotFound,
+  // NotAuthorized,
+  BadRequest,
+} from '../../../libs/api-v3/errors';
+import amzLib from '../../../libs/api-v3/amazonPayments';
+import { authWithHeaders } from '../../../middlewares/api-v3/auth';
+var payments = require('./index');
+
+let api = {};
+
+/**
+ * @api {post} /api/v3/payments/amazon/verifyAccessToken verify access token
+ * @apiVersion 3.0.0
+ * @apiName AmazonVerifyAccessToken
+ * @apiGroup Payments
+ * @apiParam {string} access_token the access token
+ * @apiSuccess {} empty
+ **/
+api.verifyAccessToken = {
+  method: 'POST',
+  url: '/payments/amazon/verifyAccessToken',
+  async handler (req, res) {
+    await amzLib.getTokenInfo(req.body.access_token)
+    .then(() => {
+      res.respond(200, {});
+    }).catch((error) => {
+      throw new BadRequest(error.body.error_description);
+    });
+  },
+};
+
+/**
+ * @api {post} /api/v3/payments/amazon/createOrderReferenceId create order reference id
+ * @apiVersion 3.0.0
+ * @apiName AmazonCreateOrderReferenceId
+ * @apiGroup Payments
+ * @apiParam {string} billingAgreementId billing agreement id
+ * @apiSuccess {object} object containing { orderReferenceId }
+ **/
+api.createOrderReferenceId = {
+  method: 'POST',
+  url: '/payments/amazon/createOrderReferenceId',
+  // middlewares: [authWithHeaders()],
+  async handler (req, res) {
+
+    try {
+      let response = await amzLib.createOrderReferenceId({
+        Id: req.body.billingAgreementId,
+        IdType: 'BillingAgreement',
+        ConfirmNow: false,
+        AWSAccessKeyId: 'something',
```

Comment:
This looks to be required now by amazon? I get the specific error 'awsaccesskeyid is required'